server/src/main/java/org/opensearch/common/util/FeatureFlags.java

```diff
@@ -12,43 +12,57 @@
 import org.opensearch.common.settings.Setting.Property;
 import org.opensearch.common.settings.Settings;
 
-import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.locks.ReentrantLock;
 
 /**
- * Utility class to manage feature flags. Feature flags are system properties that must be set on the JVM.
- * These are used to gate the visibility/availability of incomplete features. For more information, see
+ * Feature flags are used to gate the visibility/availability of incomplete features. For more information, see
  * https://featureflags.io/feature-flag-introduction/
- *
+ * Due to their specific use case, feature flag settings have several additional properties enforced by convention and code:
+ * - Feature flags are boolean settings.
+ * - Feature flags are static settings.
+ * - Feature flags are globally available.
+ * - Feature flags are configurable by JVM system properties with setting key.
  * @opensearch.internal
  */
 public class FeatureFlags {
+    // Prefixes public for testing
+    public static final String OS_EXPERIMENTAL_PREFIX = "opensearch.experimental.";
+    public static final String FEATURE_FLAG_PREFIX = OS_EXPERIMENTAL_PREFIX + "feature.";
```

Comment:
nit:
`OS_EXPERIMENTAL_PREFIX` can be private?
`FEATURE_FLAG_PREFIX` can be package private for testing?